client/landing/jetpack-cloud/sections/backups/main.jsx

```diff
@@ -24,64 +25,80 @@ import QuerySitePurchases from 'components/data/query-site-purchases';
+	onDateChange = date => {
+		this.setState( { selectedDate: date } );
+	};
+
+	onDateRangeSelection = () => {
+		//todo: go to the log activity view
+	};
 
-	hasRealtimeBackups = () =>
-		this.props.sitePurchases &&
-		!! this.props.sitePurchases.filter(
-			purchase => 'jetpack_backup_realtime' === purchase.productSlug
-		).length;
+	// hasRealtimeBackups = () =>
+	// 	this.props.sitePurchases &&
+	// 	!! this.props.sitePurchases.filter(
+	// 		purchase => 'jetpack_backup_realtime' === purchase.productSlug
+	// 	).length;
 
 	render() {
-		const { allowRestore, logs, moment, siteId, siteSlug } = this.props;
-		const { selectedDateString } = this.state;
+		// const { allowRestore, logs, moment, siteId, siteSlug } = this.props;
+		const { siteId } = this.props;
 
-		const hasRealtimeBackups = this.hasRealtimeBackups();
-		const backupAttempts = getBackupAttemptsForDate( logs, selectedDateString );
-		const deltas = getDailyBackupDeltas( logs, selectedDateString );
-		const realtimeEvents = getEventsInDailyBackup( logs, selectedDateString );
+		// const hasRealtimeBackups = this.hasRealtimeBackups();
+		// const backupAttempts = getBackupAttemptsForDate( logs, selectedDateString );
+		// const deltas = getDailyBackupDeltas( logs, selectedDateString );
+		// const realtimeEvents = getEventsInDailyBackup( logs, selectedDateString );
 
 		return (
 			<div>
 				<QueryRewindState siteId={ siteId } />
 				<QuerySitePurchases siteId={ siteId } />
+
 				<DatePicker
-					onChange={ this.dateChange }
-					selectedDateString={ selectedDateString }
+					onDateChange={ this.onDateChange }
+					onDateRangeSelection={ this.onDateRangeSelection }
+					selectedDate={ this.state.selectedDate }
 					siteId={ siteId }
 				/>
-				<DailyBackupStatus
-					allowRestore={ allowRestore }
-					date={ selectedDateString }
-					backupAttempts={ backupAttempts }
-					siteSlug={ siteSlug }
-				/>
-				<BackupDelta
-					{ ...{
-						deltas,
-						backupAttempts,
-						hasRealtimeBackups,
-						realtimeEvents,
-						allowRestore,
-						moment,
-					} }
-				/>
+				{ /* Temporaly commented for this PR */ }
+				{ /*<DailyBackupStatus*/ }
+				{ /*allowRestore={ allowRestore }*/ }
+				{ /*date={ selectedDateString }*/ }
+				{ /*backupAttempts={ backupAttempts }*/ }
+				{ /*siteSlug={ siteSlug }*/ }
+				{ /*/>*/ }
+
+				{ /*<BackupDelta*/ }
+				{ /*{ ...{*/ }
+				{ /*deltas,*/ }
+				{ /*backupAttempts,*/ }
+				{ /*hasRealtimeBackups,*/ }
+				{ /*realtimeEvents,*/ }
+				{ /*allowRestore,*/ }
+				{ /*moment,*/ }
+				{ /*} }*/ }
+				{ /*/>*/ }
 			</div>
 		);
 	}
 }
 
 export default connect( state => {
 	const siteId = getSelectedSiteId( state );
-	const logs = siteId && requestActivityLogs( siteId, emptyFilter );
-	const rewind = getRewindState( state, siteId );
-	const sitePurchases = siteId && getSitePurchases( state, siteId );
 
+	//The section require a valid site, if not, redirect to backups
```

Comment:
this sort of re-routing based on a site not being selected should be done[ in the controller function that handles rendering this Component](https://github.com/Automattic/wp-calypso/blob/3edeb919890e870a2528256b4c73fd4eda5134bd/client/landing/jetpack-cloud/sections/backups/controller.js#L21). I know that `siteId && ` is not the most elegant solution but it is basically a Calypso standard at this point

Reply:
Yeah, before Enej approach we needed to do that (better in a middleware), but now we don't need to check the `siteId` just because we are using the middleware `siteSelection` from Calypso, so we can remove this check now.